src/modules/LoginPage.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
+import { useLogin, useNotify, useGetList, useCreate, Button, Show, Labeled, TextField } from 'react-admin';
+import { Card, Stack } from '@mui/material';
+import LoginIcon from '@mui/icons-material/Login';
+
+const MyLoginPage = () => {
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const login = useLogin();
+    const notify = useNotify();
+    const [create] = useCreate();
+
+    const { data, total, isLoading } = useGetList('users');
+    const handleSubmit = (e: any) => {
+        e.preventDefault();
+        // will call authProvider.login({ email, password })
+        const userData = data?.filter((user: any) => user.name === username && user.password === password)
```

Comment:
the above line should be `find`, not `filter`. Right?

1. We want the loop to drop out as soon as a match is found
2. Our handling is simpler if the value is a valid user, or undefined.

Reply:
Aah, second note on this code. I'm pretty sure the username/password checking should be done in the `login` method of the `authProvider` - rather than having the logic here in the page code.

Reply:
If I want to move it to `authProvider` then I cannot use the `useGetList` hook(RA)  for retrieving data from `DataProvider`. 

In my understanding, this works if:
1. we have an actual API
2. or we bypass the default functions of `ra-data-local-storage` and fetch data from `localStorage` 

Thoughts @IanMayo  ?

Reply:
Would ra-data-localforage solve the issue? https://github.com/marmelab/react-admin/tree/master/packages/ra-data-localforage